website/project/model.py

```diff
@@ -2829,188 +2894,301 @@ def to_json(self):
     # {
     #   'b3k97': {
     #     'has_approved': False,
-    #     'approval_token': 'Cru7wj1Puf7DENUPFPnXSwa1rf3xPN',
-    #     'disapproval_token': 'UotzClTFOic2PYxHDStby94bCQMwJy'}
+    #     'approval_token': 'Pew7wj1Puf7DENUPFPnXSwa1rf3xPN',
+    #     'rejection_token': 'TwozClTFOic2PYxHDStby94bCQMwJy'}
     # }
     approval_state = fields.DictionaryField()
-    # One of 'pending', 'retracted', or 'cancelled'
-    state = fields.StringField(default='pending', validate=validate_retraction_state)
+    # One of 'unapproved', 'approved', or 'rejected'
+    state = fields.StringField(default='unapproved')
 
     def __repr__(self):
-        parent_registration = Node.find_one(Q('retraction', 'eq', self))
-        return ('<Retraction(parent_registration={0}, initiated_by={1}) '
-                'with _id {2}>').format(
-            parent_registration,
-            self.initiated_by,
+        return '<Sanction(end_date={1}) with _id {2}>'.format(
+            self.end_date,
             self._id
         )
 
     @property
-    def is_retracted(self):
-        return self.state == self.RETRACTED
+    def pending_approval(self):
+        return self.state == Sanction.UNAPPROVED
 
     @property
-    def pending_retraction(self):
-        return self.state == self.PENDING
+    def is_approved(self):
+        return self.state == Sanction.APPROVED
 
-    def disapprove_retraction(self, user, token):
-        """Cancels retraction if user is admin and token verifies."""
-        try:
-            if self.approval_state[user._id]['disapproval_token'] != token:
-                raise InvalidRetractionDisapprovalToken('Invalid retraction disapproval token provided.')
-        except KeyError:
-            raise PermissionsError('User must be an admin to disapprove retraction of a registration.')
+    @property
+    def is_rejected(self):
+        return self.state == Sanction.REJECTED
 
-        self.state = self.CANCELLED
-        parent_registration = Node.find_one(Q('retraction', 'eq', self))
-        parent_registration.registered_from.add_log(
-            action=NodeLog.RETRACTION_CANCELLED,
-            params={
-                'node': parent_registration._id,
-                'retraction_id': self._id,
-            },
-            auth=Auth(user),
-            save=True,
-        )
+    def _validate_authorizer(self, user):
+        return True
 
-    def approve_retraction(self, user, token):
+    def add_authorizer(self, user, approved=False, save=False):
+        valid = self._validate_authorizer(user)
+        if valid and user._id not in self.approval_state:
+            self.approval_state[user._id] = {
+                'has_approved': approved,
+                'approval_token': security.random_string(30),
+                'rejection_token': security.random_string(30),
+            }
+            if save:
+                self.save()
+            return True
+        return False
+
+    def remove_authorizer(self, user):
```

Comment:
Since you want though so much trouble to invent a new coding term I'd feel bad not changing this.